src/model/database.cpp

```diff
@@ -301,9 +314,23 @@ PObjectBase ObjectDatabase::CreateObject( std::string classname, PObjectBase par
 			bool create = true;
 
 			// we check the number of instances
-			if (max > 0 && CountChildrenWithSameType(parent, objType) >= max)
-				create = false;
+			int count;
+			if (objType->GetName() == wxT("sizer") || classname == "wxGridBagSizer")
```

Comment:
wxFB uses only types in its datamodel for these checks so don't test against a class, test against its type.

The object type system uses one object for each type and does identity comparison between them, so don't compare them by their name, just use the name to get the object and compare these. This way you don't have to create a set of `wxString` but a set of `PObjectType`, comparisons among these are cheaper than string comparisons.

A test of a concrete type would be like `objType == GetObjectType(wxT("sizer"))`